projects/Mallard/src/screens/settings/gdpr-consent-screen.tsx

```diff
@@ -161,6 +175,9 @@ const GdprConsent = ({
                             <ThreeWaySwitch
                                 onValueChange={value => {
                                     setSetting(item.key, value)
+                                    GdprBuckets[item.key].forEach(key => {
```

Comment:
I think we should extract this so that we have `setConsent` that does this work without having to remember to do it at each call site

Reply:
Agreed - will refactor.

Reply:
Cool thank 👍